src/context/EarnManagerProvider/providers/yearn/components/YearnManager/Withdraw/WithdrawReducer.ts

```diff
@@ -0,0 +1,115 @@
+import { WithdrawValues } from 'context/EarnManagerProvider/components/Withdraw/Withdraw'
+
+import { YearnVault } from '../../../api/api'
+
+type EstimatedGas = {
+  estimatedGasCrypto?: string
+}
+
+type YearnWithdrawValues = WithdrawValues &
+  EstimatedGas & {
+    txStatus: string
+    usedGasFee: string
+  }
+
+type YearnWithdrawState = {
+  vault: YearnVault
+  userAddress: string | null
+  approve: EstimatedGas
+  withdraw: YearnWithdrawValues
+  loading: boolean
+  pricePerShare: string
+  txid: string | null
+}
+
+export const initialState: YearnWithdrawState = {
+  txid: null,
+  vault: { apy: { net_apy: 0 } } as YearnVault,
+  userAddress: null,
+  loading: false,
+  approve: {},
+  pricePerShare: '',
+  withdraw: {
+    fiatAmount: '',
+    cryptoAmount: '',
+    slippage: '',
+    txStatus: 'pending',
+    usedGasFee: ''
+  }
+}
+
+export enum YearnWithdrawActionType {
+  SET_VAULT = 'SET_VAULT',
+  // SET_APPROVE = 'SET_APPROVE',
+  SET_USER_ADDRESS = 'SET_USER_ADDRESS',
+  SET_WITHDRAW = 'SET_WITHDRAW',
+  SET_LOADING = 'SET_LOADING',
+  SET_PRICE_PER_SHARE = 'SET_PRICE_PER_SHARE',
+  SET_TXID = 'SET_TXID',
+  SET_TX_STATUS = 'SET_TX_STATUS'
+}
+
+type SetVaultAction = {
+  type: YearnWithdrawActionType.SET_VAULT
+  payload: YearnVault
+}
+
+// type SetApprove = {
+//   type: YearnWithdrawActionType.SET_APPROVE
+//   payload: EstimatedGas
+// }
+
+type SetWithdraw = {
+  type: YearnWithdrawActionType.SET_WITHDRAW
+  payload: Partial<YearnWithdrawValues>
+}
+
+type SetUserAddress = {
+  type: YearnWithdrawActionType.SET_USER_ADDRESS
+  payload: string
+}
+
+type SetLoading = {
+  type: YearnWithdrawActionType.SET_LOADING
+  payload: boolean
+}
+
+type SetPricePerShare = {
+  type: YearnWithdrawActionType.SET_PRICE_PER_SHARE
+  payload: string
+}
+
+type SetTxid = {
+  type: YearnWithdrawActionType.SET_TXID
+  payload: string
+}
+
+type YearnWithdrawActions =
+  | SetVaultAction
+  // | SetApprove
+  | SetWithdraw
+  | SetUserAddress
+  | SetLoading
+  | SetPricePerShare
+  | SetTxid
+
+export const reducer = (state: YearnWithdrawState, action: YearnWithdrawActions) => {
+  switch (action.type) {
+    case YearnWithdrawActionType.SET_VAULT:
+      return { ...state, vault: { ...state.vault, ...action.payload } }
+    // case YearnWithdrawActionType.SET_APPROVE:
```

Comment:
delete commented code unless we're going to use it immediately in the next PR?

Reply:
Done.